File: services/api-agente/src/application/usecase/usuario_use_case.py
from src.application.exceptions.usuario_nao_encontrado_exception import UsuarioNaoEncontradoException
from src.infrastructure.dataprovider.usuario_data_provider import UsuarioDataProvider


class UsuarioUseCase:

    def __init__(self, usuario_data_provider: UsuarioDataProvider):
        self.usuario_data_provider = usuario_data_provider

    def consultar_por_id(self, id: str):
        usuario = self.usuario_data_provider.consultar_por_id(id)

        if usuario is None:
            raise UsuarioNaoEncontradoException()

        return usuario

File: services/api-intermediaria/src/main/java/com/gumeinteligencia/api_intermidiaria/application/usecase/UsuarioUseCase.java
package com.gumeinteligencia.api_intermidiaria.application.usecase;

import com.gumeinteligencia.api_intermidiaria.application.exceptions.UsuarioNaoEncontradoException;
import com.gumeinteligencia.api_intermidiaria.application.gateways.UsuarioGateway;
import com.gumeinteligencia.api_intermidiaria.domain.Usuario;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Optional;

@Service
@RequiredArgsConstructor
public class UsuarioUseCase {

    private final UsuarioGateway gateway;

    public Usuario consultarPorTelefoneConectado(String telefoneConectado) {
        Optional<Usuario> usuario = gateway.consultarPorTelefoneConectado(telefoneConectado);

        if(usuario.isEmpty()) {
            throw new UsuarioNaoEncontradoException();
        }

        return usuario.get();
    }
}
